src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Configuration/ConfigUpdaterStep.cs

```diff
@@ -90,6 +94,26 @@ protected override async Task<UpgradeStepInitializeResult> InitializeImplAsync(I
                 : new UpgradeStepInitializeResult(UpgradeStepStatus.Incomplete, $"{incompleteSubSteps} config updaters need applied", SubSteps.Where(s => !s.IsDone).Max(s => s.Risk));
         }
 
+        private async Task FilterSubStepsByIsApplicableAsync(IUpgradeContext context, CancellationToken token)
+        {
+            var applicableSubSteps = new List<ConfigUpdaterSubStep>();
+            foreach (var substep in _allSteps)
+            {
+                if (await substep.IsApplicableAsync(context, token).ConfigureAwait(false))
+                {
+                    applicableSubSteps.Add(substep);
+                }
+            }
+
+            SubSteps = applicableSubSteps;
+        }
+
+        public override UpgradeStepInitializeResult Reset()
+        {
+            SubSteps = _allSteps;
```

Comment:
Should we do `_allSteps = null` here?

Reply:
No, because _allSteps is set in the ctor which won't be re-invoked when initializing the step for use with the next project. Reset should undo exactly whatever Initialize does.